src/redux-query-sync.js

```diff
@@ -122,8 +130,8 @@ function ReduxQuerySync({
             // Update location (but prevent triggering a state update).
             ignoreLocationUpdate = true
             replaceState
-                ? history.replace({search: newLocationSearchString})
-                : history.push({search: newLocationSearchString})
+                ? history.replace({pathname: location.pathname, search: newLocationSearchString})
+                : history.push({pathname: location.pathname, search: newLocationSearchString})
```

Comment:
Why is setting pathname needed, is it required in history ≤v3?

Or does it ensure that the current pathname is used, even if it was changed directly (without using `history`)?
Thinking of it, ensuring this may be a wise thing to do anyway, it might actually fix the issue discussed in #10 (an outdated, cached pathname being set) even if one does not pass the `history` object. I assume however that our `getLocation()` obtains this cached (and thus possibly outdated) location anyhow; not sure if we should change it to read from `window.location` instead. We could try test this idea after merging this PR.

Reply:
Well, history v3 doesn't memoize the path object, it just get it from the passed parameter. So to support this version of history we must provide a pathname explicitly. As for getLocation method, v4 creates one location object that is changed by the lib, but v3 dosn't provide it and just wrap window.location object by calling createLocation method, so    we have to call getCurrentLocation method just right before location using. Indeed we can use window.location directly, but I think it's not good architecturaly, because doing that we get two points of truth - one is window.location and another is history.location. Of course these two sources are the same but only for now, and I think it's better not to rely on it :)